cmd/autobumper/main.go

```diff
@@ -2,6 +2,284 @@
+	flag.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name. If not specified, uses the system default.")
+	flag.Parse()
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login is mandatory")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("--github-token is mandatory")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+	return nil
+}
+
+func call(cmd string, args ...string) error {
+	c := exec.Command(cmd, args...)
+	c.Stdout = os.Stdout
+	c.Stderr = os.Stderr
+	return c.Run()
+}
+
+func updatePR(gc github.Client, org, repo, title, body, matchTitle, source, branch string) error {
+	logrus.Info("Creating PR...")
+	n, err := updater.UpdatePR(org, repo, title, body, matchTitle, gc)
+	if err != nil {
+		return fmt.Errorf("failed to update %d: %v", n, err)
+	}
+	if n == nil {
+		pr, err := gc.CreatePullRequest(org, repo, title, body, source, branch, true)
+		if err != nil {
+			return fmt.Errorf("failed to create PR: %v", err)
+		}
+		n = &pr
+	}
+
+	logrus.Infof("PR %s/%s#%d will merge %s into %s: %s", org, repo, *n, source, branch, title)
+	return nil
+}
+
+func updateReferences() (map[string]string, error) {
+	logrus.Info("Bumping image references...")
+	filter := regexp.MustCompile(prowPrefix + "|" + testImagePrefix)
+
+	err := filepath.Walk("./cluster/ci/config/prow/openshift/", func(path string, info os.FileInfo, err error) error {
+		if strings.HasSuffix(path, ".yaml") {
+			logrus.WithField("path", path).Info("updating file ...")
+			if err := bumper.UpdateFile(path, filter); err != nil {
+				logrus.WithError(err).Errorf("Failed to update path %s.", path)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return bumper.GetReplacements(), nil
+}
+
+func getNewProwVersion(images map[string]string) string {
+	for k, v := range images {
+		if strings.HasPrefix(k, prowPrefix) {
+			return v
+		}
+	}
+	return ""
+}
+
+func makeCommitSummary(images map[string]string) string {
+	return fmt.Sprintf("Update prow to %s, and other images as necessary.", getNewProwVersion(images))
+}
+
+func makeGitCommit(user, name, email string, images map[string]string) error {
+	logrus.Info("Making git commit...")
+	if err := call("git", "add", "-A"); err != nil {
+		return fmt.Errorf("failed to git add: %v", err)
+	}
+	message := makeCommitSummary(images)
+	commitArgs := []string{"commit", "-m", message}
+	if name != "" && email != "" {
+		commitArgs = append(commitArgs, "--author", fmt.Sprintf("%s <%s>", name, email))
+	}
+	if err := call("git", commitArgs...); err != nil {
+		return fmt.Errorf("failed to git commit: %v", err)
+	}
+	logrus.Info("Pushing to remote...")
+	if err := call("git", "push", "-f", fmt.Sprintf("git@github.com:%s/test-infra.git", user), "HEAD:autobump"); err != nil {
```

Comment:
This isn't the right place to push things

Reply:
Done.
Passed as parameter when calling API.